Source/WebCore/page/Navigation.h

```diff
@@ -46,11 +45,13 @@ enum class FrameLoadType : uint8_t;
 
 // https://html.spec.whatwg.org/multipage/nav-history-apis.html#navigation-api-method-tracker
 struct NavigationAPIMethodTracker {
-    NavigationAPIMethodTracker(uint64_t id, Ref<DeferredPromise>&& committed, Ref<DeferredPromise>&& finished, JSC::JSValue&& info, RefPtr<SerializedScriptValue>&& serializedState)
+    explicit NavigationAPIMethodTracker() = default;
+
+    explicit NavigationAPIMethodTracker(uint64_t id, Ref<DeferredPromise>&& committed, Ref<DeferredPromise>&& finished, JSC::JSValue&& info, RefPtr<SerializedScriptValue>&& serializedState)
         : info(info)
         , serializedState(serializedState)
-        , committedPromise(DOMPromise::create(*committed->globalObject(), *JSC::jsCast<JSC::JSPromise*>(committed->promise())))
-        , finishedPromise(DOMPromise::create(*finished->globalObject(), *JSC::jsCast<JSC::JSPromise*>(finished->promise())))
+        , committedPromise(&committed.leakRef())
+        , finishedPromise(&finished.leakRef())
```

Comment:
Should this just use WTFMove?

Reply:
It is transferring ownership of a `Ref` to a `RefPtr`. I saw this pattern in a few places.

Reply:
Yeah, this looks like a memory leak as it stands now.

Reply:
Great, the important think is indeed that NavigationAPIMethodTracker takes ownership.

Reply:
God please don't do this lol. use `WTFMove()`. I'm very curious where you saw this "pattern".

Reply:
I don't remember, it's used ~121 times in the codebase.

This is a move constructor for RefPtr though:

```cpp
RefPtr<T, U, V>::RefPtr(Ref<X, Y>&& reference)
    : m_ptr(&reference.leakRef())
```

I understand that this wouldn't ref while my usage would though. My bad.